Lib/fontTools/designspaceLib/split.py

```diff
@@ -225,6 +226,20 @@ def maybeExpandDesignLocation(object):
                 )
             )
 
+    # TODO Trim out-of-range values? :-(
+    subDoc.axisMappings = mappings = []
+    subDocAxes = [axis.name for axis in subDoc.axes]
+    for mapping in doc.axisMappings:
+        if not all(axis in subDocAxes for axis in mapping.inputLocation.keys()):
+            continue
+        assert all(axis in subDocAxes for axis in mapping.outputLocation.keys())
```

Comment:
assert here means this should never trigger. IIUC the assumption is that if all the axis in the inputLocation are in this subDoc, then also all those in the outputLocation must be. If it's conceivable that this assertion may trigger because of result of user error, then we should provide a better error message than plain assertion. If this is simply checking inner code logic and can never trigger, then i'm fine with leaving as such.

Reply:
I think an error is more appropriate indeed.  I'm not very familiar with how axes are chosen for the split to know if it can happen or not. Help needed.

Reply:
Reading the calls to the split function, it looks like all non-discrete axes are always included in the split. So indeed an assert should be fine. Am I right @belluzj?

The only issue outstanding is to limit the range if the input range is limited. That might be tricky though as in https://github.com/fonttools/fonttools/issues/3086

Reply:
> Reading the calls to the split function, it looks like all non-discrete axes are always included in the split.

I'm wrong I think. Adding a test and making it an error.